app/presenters/shelf_list_presenter.rb

```diff
@@ -0,0 +1,125 @@
+      documents.slice(1, length)
+    end
+  end
+
+  def next
+    return if the_end?
+
+    documents.last
+  end
+
+  def previous
+    return if beginning?
+
+    documents.first
+  end
+
+  private
+
+    def documents
+      (before_items + after_items)
+    end
+
+    # @return [Array<ShelfItem>]
+    # @note The items coming before a particular call number are determined using a reverse shelf key, which means the
+    # items are returned in a "reverse" order. To create a natural reading list, we need to reverse the reversed list
+    # before adding it to our virtual shelf. If we're looking for something nearby on the shelf, and our query is an
+    # exact match for that item, mark it accordingly. Otherwise, insert a placeholder shelf item that indicates where
+    # the item would appear if it existed.
+    def before_items
+      before_list = shelf_list[:before].reverse
+
+      return before_list if nearby.blank?
+
+      if before_list.last.call_number == nearby
+        before_list.last.match = true
+      else
+        before_list << ShelfItem.new(label: "You're looking for: #{nearby}", call_number: 'None', key: nil)
+        before_list.shift
+      end
+
+      before_list.last.nearby = true
+      before_list
+    end
+
+    # @return [Array<ShelfItem>]
+    def after_items
+      shelf_list[:after]
+    end
+
+    def shelf_list
+      @shelf_list ||= ShelfList.call(shelf_list_params)
+    end
+
+    # @note There are four possible scenarios, each determined by the parameters given to the presenter, and each
+    # resulting in a specific set of arguments passed to the ShelfList service.
+    #
+    # 1) Querying from a known point forward:
+    #    Returns the number of items after a given key. The two items returned before the key are the item itself, and
+    #    the previous item which is used to construct the link to the previous shelf.
+    # 2) Querying from a known point backwards:
+    #    Returns the number of items before a given key. The one item that occurs after the key is used to construct the
+    #    link to the next shelf.
+    # 3) Looking from something nearby on the shelf, which may _or_ may not exist:
+    #    Returns a set of before and after items such that the item we're looking for occurs *third* on the shelf. This
+    #    is basically padding the results based on our minimum shelf length.
+    # 4) Starting from the very beginning:
+    #    Return only the items after the given key. A '0' query starts at the very beginning of the list.
+    def shelf_list_params
+      if starting.present?
+        { query: starting, forward_limit: length, reverse_limit: 2 }
+      elsif ending.present?
+        { query: ending, forward_limit: 1, reverse_limit: length + 1 }
+      elsif nearby.present?
+        { query: nearby, forward_limit: (length - MIN) + 2, reverse_limit: MIN }
+      else
+        { query: '0', forward_limit: length + 1, reverse_limit: 0 }
+      end
+    end
+
+    def beginning?
```

Comment:
@awead does it make sense to rename this one as `the_start` 🤔

Reply:
It's returning true/false. I initially had something like `at_the_beginning?` but that was too long.

Reply:
`the_start?` because `the_end?` is what I was thinking, just forgot to add the `?`  to them in my comment.

Reply:
if you think it sounds weird, it is OK no need to change.

Reply:
How about first?/last?

Reply:
yes, loved it! more clear and distinct 👌🏻